homeassistant/components/samsungtv/config_flow.py

```diff
@@ -95,81 +156,79 @@ async def async_step_import(self, user_input=None):
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         if user_input is not None:
-            ip_address = await self.hass.async_add_executor_job(
-                _get_ip, user_input[CONF_HOST]
-            )
-
-            await self.async_set_unique_id(ip_address)
-            self._abort_if_unique_id_configured()
-
-            self._host = user_input.get(CONF_HOST)
-            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
-            self._name = user_input.get(CONF_NAME)
+            try:
+                self._host = await self.hass.async_add_executor_job(
+                    socket.gethostbyname, user_input[CONF_HOST]
+                )
+            except socket.gaierror as err:
+                raise data_entry_flow.AbortFlow(RESULT_UNKNOWN_HOST) from err
+            self._name = user_input[CONF_NAME]
             self._title = self._name
 
-            result = await self.hass.async_add_executor_job(self._try_connect)
+            await self.hass.async_add_executor_job(self._try_connect)
+
+            await self._async_set_device_unique_id(raise_on_progress=False)
 
-            if result != RESULT_SUCCESS:
-                return self.async_abort(reason=result)
             return self._get_entry()
 
         return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
 
-    async def async_step_ssdp(self, discovery_info):
-        """Handle a flow initialized by discovery."""
-        host = urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname
-        ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+    async def async_step_ssdp(self, discovery_info: DiscoveryInfoType):
+        """Handle a flow initialized by ssdp discovery."""
+        self._host = urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname
+
+        LOGGER.debug("Found Samsung device via ssdp at %s", self._host)
+        await self._async_set_device_unique_id()
 
-        self._host = host
-        self._ip = self.context[CONF_IP_ADDRESS] = ip_address
         self._manufacturer = discovery_info.get(ATTR_UPNP_MANUFACTURER)
-        self._model = discovery_info.get(ATTR_UPNP_MODEL_NAME)
-        self._name = f"Samsung {self._model}"
-        self._id = discovery_info.get(ATTR_UPNP_UDN)
+        if not self._model:
+            self._model = discovery_info.get(ATTR_UPNP_MODEL_NAME)
```

Comment:
```suggestion
        self._model = discovery_info.get(ATTR_UPNP_MODEL_NAME)
```

No need to check if its already set since it can't be since this is the start of the flow

Reply:
I guess its possible that it could be missing from device info?